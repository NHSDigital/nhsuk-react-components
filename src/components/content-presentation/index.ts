export * from './details/index.js';
export * from './do-and-dont-list/index.js';
export * from './hero/index.js';
export * from './icons/index.js';
export * from './images/index.js';
export * from './inset-text/index.js';
export * from './notification-banner/index.js';
export * from './panel/index.js';
export * from './summary-list/index.js';
export * from './table/index.js';
export * from './tabs/index.js';
export * from './tag/index.js';
export * from './warning-callout/index.js';
