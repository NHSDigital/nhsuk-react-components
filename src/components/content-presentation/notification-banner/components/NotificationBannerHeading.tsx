import { HeadingLevel, type HeadingLevelProps } from '#components';
import type { FC } from 'react';

export type NotificationBannerHeadingProps = HeadingLevelProps;

const NotificationBannerHeadingComponent: FC<NotificationBannerHeadingProps> = ({
  children,
  headingLevel = 'h3',
  ...rest
}) => (
  <HeadingLevel
    className="nhsuk-notification-banner__heading"
    headingLevel={headingLevel}
    {...rest}
  >
    {children}
  </HeadingLevel>
);

NotificationBannerHeadingComponent.displayName = 'NotificationBanner.Heading';

export const NotificationBannerHeading = Object.assign(NotificationBannerHeadingComponent);
