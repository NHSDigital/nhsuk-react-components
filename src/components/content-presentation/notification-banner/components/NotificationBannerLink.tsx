import type { AsElementLink } from '#util/types/index.js';
import { forwardRef } from 'react';
import classNames from 'classnames';

export type NotificationBannerLinkProps = AsElementLink<HTMLAnchorElement>;

const NotificationBannerLinkComponent = forwardRef<HTMLAnchorElement, NotificationBannerLinkProps>(
  ({ children, className, asElement: Element = 'a', ...rest }, forwardedRef) => (
    <Element
      className={classNames('nhsuk-notification-banner__link', className)}
      ref={forwardedRef}
      {...rest}
    >
      {children}
    </Element>
  ),
);

NotificationBannerLinkComponent.displayName = 'NotificationBanner.Link';

export const NotificationBannerLink = Object.assign(NotificationBannerLinkComponent);
