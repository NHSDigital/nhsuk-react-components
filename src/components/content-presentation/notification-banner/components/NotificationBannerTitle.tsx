import { HeadingLevel, type HeadingLevelProps } from '#components/utils/HeadingLevel.js';
import type { FC } from 'react';

export interface NotificationBannerTitleProps extends HeadingLevelProps {
  success?: boolean;
}

export const NotificationBannerTitle: FC<NotificationBannerTitleProps> = ({
  children,
  headingLevel = 'h2',
  id = 'nhsuk-notification-banner-title',
  success,
  ...rest
}) => (
  <div className="nhsuk-notification-banner__header">
    <HeadingLevel
      className="nhsuk-notification-banner__title"
      headingLevel={headingLevel}
      id={id}
      {...rest}
    >
      {children || (success ? 'Success' : 'Important')}
    </HeadingLevel>
  </div>
);

NotificationBannerTitle.displayName = 'NotificationBanner.Title';
