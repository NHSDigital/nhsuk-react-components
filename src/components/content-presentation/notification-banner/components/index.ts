export * from './NotificationBannerHeading.js';
export * from './NotificationBannerLink.js';
export * from './NotificationBannerTitle.js';
