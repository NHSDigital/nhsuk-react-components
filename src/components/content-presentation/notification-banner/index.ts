export * from './components/index.js';
export * from './NotificationBanner.js';
