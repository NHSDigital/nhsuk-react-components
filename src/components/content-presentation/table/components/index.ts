export * from './TableBody';
export * from './TableCaption';
export * from './TableCell';
export * from './TableContainer';
export * from './TableHead';
export * from './TablePanel';
export * from './TableRow';
