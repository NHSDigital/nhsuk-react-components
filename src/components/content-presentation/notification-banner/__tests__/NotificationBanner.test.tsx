import { renderClient, renderServer } from '#util/components';
import { NotificationBanner } from '#components/content-presentation/notification-banner';
import { createRef } from 'react';
import { NotificationBannerLink } from '#components/content-presentation/notification-banner/components';

describe('NotificationBanner', () => {
  it('matches snapshot', async () => {
    const { container } = await renderClient(
      <NotificationBanner>
        <NotificationBanner.Heading>
          The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
        </NotificationBanner.Heading>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot success', async () => {
    const { container } = await renderClient(
      <NotificationBanner success>
        <NotificationBanner.Heading>Patient record updated</NotificationBanner.Heading>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot custom title', async () => {
    const { container } = await renderClient(
      <NotificationBanner title={'Upcoming maintenance'}>
        <NotificationBanner.Heading>
          The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
        </NotificationBanner.Heading>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot custom title html', async () => {
    const { container } = await renderClient(
      <NotificationBanner>
        <NotificationBanner.Title>
          <strong>Very</strong> important information
        </NotificationBanner.Title>
        <NotificationBanner.Heading>
          The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
        </NotificationBanner.Heading>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with link in heading', async () => {
    const { container } = await renderClient(
      <NotificationBanner>
        <NotificationBanner.Heading>
          You have 7 days left to send your application.{' '}
          <NotificationBanner.Link href="#">View application</NotificationBanner.Link>.
        </NotificationBanner.Heading>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with link in body', async () => {
    const { container } = await renderClient(
      <NotificationBanner>
        <NotificationBanner.Heading>Patient record updated</NotificationBanner.Heading>
        <p>
          Contact{' '}
          <NotificationBanner.Link href="#">example@department.nhs.uk</NotificationBanner.Link> if
          you think there&#39;s a problem.
        </p>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with no sub elements', async () => {
    const { container } = await renderClient(
      <NotificationBanner>
        The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with list', async () => {
    const { container } = await renderClient(
      <NotificationBanner>
        <NotificationBanner.Heading>4 files uploaded</NotificationBanner.Heading>
        <ul>
          <li>
            <NotificationBannerLink>government-strategy.pdf</NotificationBannerLink>
          </li>
          <li>
            <NotificationBannerLink>government-strategy-v2.pdf</NotificationBannerLink>
          </li>
          <li>
            <NotificationBannerLink>government-strategy-v3-FINAL.pdf</NotificationBannerLink>
          </li>
          <li>
            <NotificationBannerLink>government-strategy-v4-FINAL-v2.pdf</NotificationBannerLink>
          </li>
        </ul>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with long heading', async () => {
    const { container } = await renderClient(
      <NotificationBanner>
        <NotificationBanner.Heading>
          The patient record was withdrawn on 7 March 2014, before being sent in, sent back,
          queried, lost, found, subjected to public inquiry, lost again, and finally buried in soft
          peat for three months and recycled as firelighters.
        </NotificationBanner.Heading>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with lots of content', async () => {
    const { container } = await renderClient(
      <NotificationBanner>
        <NotificationBanner.Heading>
          Check if you need to apply the reverse charge to this application
        </NotificationBanner.Heading>
        <p>
          You will have to apply the{' '}
          <NotificationBanner.Link href="#">reverse charge</NotificationBanner.Link> if the
          applicant supplies any of these services:
        </p>
        <ul>
          <li>
            constructing, altering, repairing, extending, demolishing or dismantling buildings or
            structures (whether permanent or not), including offshore installation services
          </li>
          <li>
            constructing, altering, repairing, extending, demolishing of any works forming, or
            planned to form, part of the land, including (in particular) walls, roadworks, power
            lines, electronic communications equipment, aircraft runways, railways, inland
            waterways, docks and harbours
          </li>
        </ul>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot (via server)', async () => {
    const { container } = await renderServer(
      <NotificationBanner>
        <NotificationBanner.Heading>
          The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
        </NotificationBanner.Heading>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot success (via server)', async () => {
    const { container } = await renderServer(
      <NotificationBanner success>
        <NotificationBanner.Heading>Patient record updated</NotificationBanner.Heading>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot custom title (via server)', async () => {
    const { container } = await renderServer(
      <NotificationBanner title={'Upcoming maintenance'}>
        <NotificationBanner.Heading>
          The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
        </NotificationBanner.Heading>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot custom title html (via server)', async () => {
    const { container } = await renderServer(
      <NotificationBanner>
        <NotificationBanner.Title id={'custom-title'}>
          <strong>Very</strong> important information
        </NotificationBanner.Title>
        <NotificationBanner.Heading>
          The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
        </NotificationBanner.Heading>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with link in heading (via server)', async () => {
    const { container } = await renderServer(
      <NotificationBanner>
        <NotificationBanner.Heading>
          You have 7 days left to send your application.{' '}
          <NotificationBanner.Link href="#">View application</NotificationBanner.Link>.
        </NotificationBanner.Heading>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with link in body (via server)', async () => {
    const { container } = await renderServer(
      <NotificationBanner>
        <NotificationBanner.Heading>Patient record updated</NotificationBanner.Heading>
        <p>
          Contact{' '}
          <NotificationBanner.Link href="#">example@department.nhs.uk</NotificationBanner.Link> if
          you think there&#39;s a problem.
        </p>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with no sub elements (via server)', async () => {
    const { container } = await renderServer(
      <NotificationBanner>
        The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with list (via server)', async () => {
    const { container } = await renderServer(
      <NotificationBanner>
        <NotificationBanner.Heading>4 files uploaded</NotificationBanner.Heading>
        <ul>
          <li>
            <NotificationBannerLink>file.pdf</NotificationBannerLink>
          </li>
          <li>
            <NotificationBannerLink>file-v2.pdf</NotificationBannerLink>
          </li>
          <li>
            <NotificationBannerLink>file-v3-FINAL.pdf</NotificationBannerLink>
          </li>
          <li>
            <NotificationBannerLink>file-v4-FINAL-v2.pdf</NotificationBannerLink>
          </li>
        </ul>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with long heading (via server)', async () => {
    const { container } = await renderServer(
      <NotificationBanner>
        <NotificationBanner.Heading>
          The patient record was withdrawn on 7 March 2014, before being sent in, sent back,
          queried, lost, found, subjected to public inquiry, lost again, and finally buried in soft
          peat for three months and recycled as firelighters.
        </NotificationBanner.Heading>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with lots of content (via server)', async () => {
    const { container } = await renderServer(
      <NotificationBanner>
        <NotificationBanner.Heading>
          Check if you need to apply the reverse charge to this application
        </NotificationBanner.Heading>
        <p>
          You will have to apply the{' '}
          <NotificationBanner.Link href="#">reverse charge</NotificationBanner.Link> if the
          applicant supplies any of these services:
        </p>
        <ul>
          <li>
            constructing, altering, repairing, extending, demolishing or dismantling buildings or
            structures (whether permanent or not), including offshore installation services
          </li>
          <li>
            constructing, altering, repairing, extending, demolishing of any works forming, or
            planned to form, part of the land, including (in particular) walls, roadworks, power
            lines, electronic communications equipment, aircraft runways, railways, inland
            waterways, docks and harbours
          </li>
        </ul>
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    expect(container).toMatchSnapshot();
  });

  it('forwards refs', async () => {
    const ref = createRef<HTMLDivElement>();

    const { modules } = await renderClient(
      <NotificationBanner ref={ref}>
        The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    const [notificationBannerEl] = modules;

    expect(ref.current).toBe(notificationBannerEl);
    expect(ref.current).toHaveClass('nhsuk-notification-banner');
  });

  it('has default role and autofocus', async () => {
    const { modules } = await renderClient(
      <NotificationBanner>
        The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    const [notificationBannerEl] = modules;

    expect(notificationBannerEl?.getAttribute('role')).toBe('region');
    expect(notificationBannerEl?.dataset?.disableAutoFocus).toBe(undefined);
  });

  it('has alert role', async () => {
    const { modules } = await renderClient(
      <NotificationBanner success>
        The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    const [notificationBannerEl] = modules;

    expect(notificationBannerEl?.getAttribute('role')).toBe('alert');
  });

  it('has disabled autofocus', async () => {
    const { modules } = await renderClient(
      <NotificationBanner disableAutoFocus>
        The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
      </NotificationBanner>,
      { className: 'nhsuk-notification-banner' },
    );

    const [notificationBannerEl] = modules;

    expect(notificationBannerEl?.dataset?.disableAutoFocus).toBe('true');
  });
});
