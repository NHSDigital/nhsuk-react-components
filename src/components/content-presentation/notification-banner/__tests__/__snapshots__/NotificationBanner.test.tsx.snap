// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NotificationBanner matches snapshot (via server) 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
      </h3>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    data-nhsuk-notification-banner-init=""
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
      </h3>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot custom title (via server) 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Upcoming Maintenance
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
      </h3>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot custom title 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    data-nhsuk-notification-banner-init=""
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Upcoming Maintenance
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
      </h3>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot custom title html (via server) 1`] = `
<div>
  <div
    aria-labelledby="custom-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="custom-title"
      >
        <strong>
          Very
        </strong>
         important information
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
      </h3>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot custom title html 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    data-nhsuk-notification-banner-init=""
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        <strong>
          Very
        </strong>
         important information
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
      </h3>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot success (via server) 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner nhsuk-notification-banner--success"
    data-module="nhsuk-notification-banner"
    role="alert"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Success
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        Patient record updated
      </h3>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot success 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner nhsuk-notification-banner--success"
    data-module="nhsuk-notification-banner"
    data-nhsuk-notification-banner-init=""
    role="alert"
    tabindex="-1"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Success
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        Patient record updated
      </h3>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot with link in body (via server) 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        Patient record updated
      </h3>
      <p>
        Contact
        <!-- -->
         
        <a
          class="nhsuk-notification-banner__link"
          href="#"
        >
          example@department.nhs.uk
        </a>
         if you think there's a problem.
      </p>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot with link in body 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    data-nhsuk-notification-banner-init=""
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        Patient record updated
      </h3>
      <p>
        Contact
         
        <a
          class="nhsuk-notification-banner__link"
          href="#"
        >
          example@department.nhs.uk
        </a>
         if you think there's a problem.
      </p>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot with link in heading (via server) 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        You have 7 days left to send your application.
        <!-- -->
         
        <a
          class="nhsuk-notification-banner__link"
          href="#"
        >
          View application
        </a>
        .
      </h3>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot with link in heading 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    data-nhsuk-notification-banner-init=""
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        You have 7 days left to send your application.
         
        <a
          class="nhsuk-notification-banner__link"
          href="#"
        >
          View application
        </a>
        .
      </h3>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot with list (via server) 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        4 files uploaded
      </h3>
      <ul>
        <li>
          <a
            class="nhsuk-notification-banner__link"
          >
            file.pdf
          </a>
        </li>
        <li>
          <a
            class="nhsuk-notification-banner__link"
          >
            file-v2.pdf
          </a>
        </li>
        <li>
          <a
            class="nhsuk-notification-banner__link"
          >
            file-v3-FINAL.pdf
          </a>
        </li>
        <li>
          <a
            class="nhsuk-notification-banner__link"
          >
            file-v4-FINAL-v2.pdf
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot with list 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    data-nhsuk-notification-banner-init=""
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        4 files uploaded
      </h3>
      <ul>
        <li>
          <a
            class="nhsuk-notification-banner__link"
          >
            government-strategy.pdf
          </a>
        </li>
        <li>
          <a
            class="nhsuk-notification-banner__link"
          >
            government-strategy-v2.pdf
          </a>
        </li>
        <li>
          <a
            class="nhsuk-notification-banner__link"
          >
            government-strategy-v3-FINAL.pdf
          </a>
        </li>
        <li>
          <a
            class="nhsuk-notification-banner__link"
          >
            government-strategy-v4-FINAL-v2.pdf
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot with long heading (via server) 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        The patient record was withdrawn on 7 March 2014, before being sent in, sent back, queried, lost, found, subjected to public inquiry, lost again, and finally buried in soft peat for three months and recycled as firelighters.
      </h3>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot with long heading 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    data-nhsuk-notification-banner-init=""
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        The patient record was withdrawn on 7 March 2014, before being sent in, sent back, queried, lost, found, subjected to public inquiry, lost again, and finally buried in soft peat for three months and recycled as firelighters.
      </h3>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot with lots of content (via server) 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        Check if you need to apply the reverse charge to this application
      </h3>
      <p>
        You will have to apply the
        <!-- -->
         
        <a
          class="nhsuk-notification-banner__link"
        >
          reverse charge
        </a>
         if the applicant supplies any of these services:
      </p>
      <ul>
        <li>
          constructing, altering, repairing, extending, demolishing or dismantling buildings or structures (whether permanent or not), including offshore installation services
        </li>
        <li>
          constructing, altering, repairing, extending, demolishing of any works forming, or planned to form, part of the land, including (in particular) walls, roadworks, power lines, electronic communications equipment, aircraft runways, railways, inland waterways, docks and harbours
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot with lots of content 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    data-nhsuk-notification-banner-init=""
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      <h3
        class="nhsuk-notification-banner__heading"
      >
        Check if you need to apply the reverse charge to this application
      </h3>
      <p>
        You will have to apply the
         
        <a
          class="nhsuk-notification-banner__link"
        >
          reverse charge
        </a>
         if the applicant supplies any of these services:
      </p>
      <ul>
        <li>
          constructing, altering, repairing, extending, demolishing or dismantling buildings or structures (whether permanent or not), including offshore installation services
        </li>
        <li>
          constructing, altering, repairing, extending, demolishing of any works forming, or planned to form, part of the land, including (in particular) walls, roadworks, power lines, electronic communications equipment, aircraft runways, railways, inland waterways, docks and harbours
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot with no sub elements (via server) 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
    </div>
  </div>
</div>
`;

exports[`NotificationBanner matches snapshot with no sub elements 1`] = `
<div>
  <div
    aria-labelledby="nhsuk-notification-banner-title"
    class="nhsuk-notification-banner"
    data-module="nhsuk-notification-banner"
    data-nhsuk-notification-banner-init=""
    role="region"
  >
    <div
      class="nhsuk-notification-banner__header"
    >
      <h2
        class="nhsuk-notification-banner__title"
        id="nhsuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="nhsuk-notification-banner__content"
    >
      The service will be unavailable from 8pm to 9pm on Thursday 1 January 2025.
    </div>
  </div>
</div>
`;
